### ISSUE TITLE
dependsOn has no effect when the parent field is inside flexible

### ISSUE THREAD

**Issue by fenestron** [whitecube/nova-flexible-content#341]
## Code example:
```
Text::make('Recipient')
    ->readonly()
    ->dependsOn(
        ['type'],
        function (Text $field, NovaRequest $request, FormData $formData) {
            if ($formData->type === 'gift') {
                $field->readonly(false)->rules(['required', 'email']);
            }
        }
    ),

Flexible::make('Content')
    ->addLayout('Simple content section', 'wysiwyg', [
        Text::make('Recipient')
            ->readonly()
            ->dependsOn(
                ['type'],
                function (Text $field, NovaRequest $request, FormData $formData) {
                    if ($formData->type === 'gift') {
                        $field->readonly(false)->rules(['required', 'email']);
                    }
                }
            ),

        Select::make('Purchase Type', 'type')
            ->options([
                'personal' => 'Personal',
                'gift' => 'Gift',
            ])
            ->default('personal'),
    ]),
```

## Field behavior:

![Laravel - Create Test](https://user-images.githubusercontent.com/9680621/163214723-0680a687-34cb-4e61-a8fb-a37819d498cb.gif)

P.S. Example taken from [documentation](https://nova.laravel.com/docs/4.0/resources/fields.html#dependent-fields)

**Comment by voidgraphics** [comment]
Thanks for the report!

**Comment by stepanenko3** [comment]
The Slug field does not work by analogy

**Comment by ianrobertsFF** [comment]
To be fair the slug field never worked even under Nova 3. I had to Make my own slug field that checked if it had a Flexible Content key prefix to its id and then apply that prefix to the field it was checking.

Biggest issue really is that the temporary keys are generated wholly in javascript, and not from and AJAX call to the layout. If it were the latter you could account for all of this in your layout and never have to touch Nova's Vue files.

**Comment by ianrobertsFF** [comment]
I could, but I'm not really a developer, more developer adjacent infra background working on a personal project. I have a Nova 4 License so ill spin up a test site and take a look over this long weekend and see if I can pull something together. 

If anyone else would rather take a stab at it your code is probably going to be better than mine!

**Comment by ianrobertsFF** [comment]
It might actually be simpler than that.

In Nova 4 formFields have a prop of `formUniqueId`

This is checked when registering for events:

`getFieldAttributeChangeEventName(attribute) {
      return this.hasFormUniqueId === true
        ? `${this.formUniqueId}-${attribute}-change`
        : `${attribute}-change`
    },`

It might be as simple as passing `group.key` as `:form-unique-id` in the component in  `FormGroup.vue`

Don't have my test install set up yet. But will test this first.

**Comment by ianrobertsFF** [comment]
Been playing with this a little more this evening.

Have fixed the validationKey being overwritten with a ref for the fields and a mounted hook to check when syncedField is updated and force the syncedField validationKey back to the custom one needed by Flexible Content.

https://github.com/ianrobertsFF/nova-flexible-content/commit/ccf0a9abad7d474f976ff39222b08223e58accca

**Comment by ianrobertsFF** [comment]
@voidgraphics 

Any movement on this one? Would like to abandon my repo on this. If you have any notes on my solution I'm happy to take another look at it

**Comment by ianrobertsFF** [comment]
Closed the PR, will implement in a fork

**Comment by Saifallak** [comment]
> Closed the PR, will implement in a fork

any sol. so far?

**Comment by ianrobertsFF** [comment]
Yeah, as I said I implemented in a fork, although that fork is a little behind on merging the main repos commits.

**Comment by Saifallak** [comment]
> Yeah, as I said I implemented in a fork, although that fork is a little behind on merging the main repos commits.

fork url please? and branch ?

**Comment by ianrobertsFF** [comment]
> > Yeah, as I said I implemented in a fork, although that fork is a little behind on merging the main repos commits.
> 
> fork url please? and branch ?

https://github.com/Formfeed-UK/nova-flexible-content

If you use it, please don't expect immediate feature merges, bug fixes should be pretty quick though.
